devel/kchmviewer/files/patch-use-webkit-and-dbus.diff

```diff
@@ -0,0 +1,71 @@
+diff --git a/src/src.pro b/src/src.pro
+index 2524b5e..f9e7098 100644
+--- a/src/src.pro
++++ b/src/src.pro
+@@ -40,7 +40,7 @@ SOURCES += config.cpp \
+     textencodings.cpp \
+     treeitem_toc.cpp \
+     treeitem_index.cpp
+-LIBS += -lchm -lzip
++#LIBS += -lchm -lzip
+ TARGET = ../bin/kchmviewer
+ CONFIG += threads \
+     warn_on \
+@@ -81,7 +81,7 @@ macx: {
+     HEADERS += kchmviewerapp.h
+     SOURCES += kchmviewerapp.cpp
+     QMAKE_INFO_PLIST=resources/Info.plist
+-    QMAKE_POST_LINK += cp resources/*.icns ${DESTDIR}/kchmviewer.app/Contents/Resources;
++    ICON = resources/kchmviewer.icns
+     LIBS += ../lib/libebook/libebook.a
+     POST_TARGETDEPS += ../lib/libebook/libebook.a
+ }
+@@ -102,7 +102,7 @@ win32-*: {
+     LIBS += -lwsock32 -loleaut32
+ }
+ 
+-unix:!macx: {
++unix: {
+ 
+     QT += dbus
+     HEADERS += dbus_interface.h
+@@ -112,6 +111,9 @@ unix: {
+     POST_TARGETDEPS += ../lib/libebook/libebook.a
+ }
+ 
++LIBS += -L@PREFIX@/lib -lchm -L@PREFIX@/lib/ -lzip
```

Comment:
This also sounds pretty strange to me. The only problem I even remember was interference between the already installed version of a library and the version being compiled. From what I remember the universal recipe was the order of `-L/path/to` flags and sometimes `-L/opt/local/lib` would come in front of `-L../lib`.

Usually projects had problems when they switched to CMake. In `configure`-based installation they would add `-L../lib` and alike at the front of the build command. In CMake they would add it at the end and then the already installed library had preference due to `-L/opt/local/lib` coming from another dependency. But this is fully orthogonal.

I would in fact like to see examples where this fails.

Maybe one indeed needs to patch the file by adding `-L@PREFIX@/lib`, but I cannot imagine that one would have to reshuffle the order and add the second copy of `/opt/local/lib` to the search path. After all at some point you would end up with `-L@PREFIX@/lib/////////`. This cannot be.